File: Learn/5. Stacks/stackUsingArray.java
import java.util.Arrays;
import java.util.EmptyStackException;

public class stackUsingArray {

    static class Stack {
        private Integer[] stackArray;
        private Integer stackSize;
        private Integer index = 0;

        public Stack(Integer size) {
            stackArray = new Integer[size];
            stackSize = size;
        }

        private boolean isStackFull(Integer stackSize, Integer index) {

            if (index == stackSize) {
                return true;
            }
            return false;
        }

        public void push(Integer value) {
            if (isStackFull(stackSize, index) == true) {
                throw new StackOverflowError("Stack is full");
            }
            stackArray[index] = value;
            index += 1;

        }

        public Integer pop() {
            if (isEmpty()) {
                throw new EmptyStackException();
            }
            Integer returnValue = stackArray[index - 1];
            stackArray[index - 1] = null;
            index -= 1;
            return returnValue;
        }

        public Integer peek() {
            if (isEmpty()) {
                throw new EmptyStackException();
            }
            return stackArray[index - 1];
        }

        public boolean isEmpty() {
            if (index == 0) {
                return true;
            }
            return false;
        }

        public Integer[] getStack() {
            return stackArray;
        }
    }

    public static void main(String[] args) {
        Stack test = new Stack(5);
        test.push(5);
        test.push(4);
        test.push(3);
        test.push(2);
        test.push(1);
        System.out.println(test.pop());
        System.out.println(test.peek());
        System.out.println(Arrays.toString(test.getStack()));
    }
}

File: 5. Stacks/stackUsingArray.js
class Stack {
    constructor(size){
        this.size = size;
        this.index = 0;
        this.stack = [];
    }
    push = (element) =>{
        if(this.isFull()){
            throw 'Stack is Full!'
        }else{
            this.stack[this.index] = element;
            this.index+=1;
        }
    }
    pop = () =>{
        if(this.isEmpty()){
            throw 'Stack is empty';
        }else{
            let returnElement = this.stack[this.index - 1];
            this.stack[this.index -1] = null;
            this.index -=1;
            return returnElement;
        }
    }
    peek = () =>{
        if(this.isEmpty()){
            throw 'Stack is empty';
        }else{
            return this.stack[this.index - 1];
        }
    }
    isFull = () =>{
         if(this.index == this.size){
             return true
         }
         return false;
     }

     isEmpty = () =>{
         if(this.index == 0){
             return true;
         }
         return false;
     }

     clear = () =>{
         for(let i = 0; i< this.size;i++){
            this.stack[i] = null;
         }
     }
     print = () =>{
         console.log(this.stack);
     }
}

let test = new Stack(3);
test.push(1);
test.push(2);
test.push(3);
test.print();